### PR TITLE
@alloy => [Hero Carousel] fix auto-advance and do not lose position on view change

### LINKED ISSUE #746: [Featured Carousel] Maintain current slide when navigating away/back
From @robertlenne:
> When I go one view in from home-view and then back the homepage banner jumps back to first position which looks glitchy. Think the homepage carousel should just stay on the slide it's on. 

To replicate: Open the Artsy app, wait for the featured carousel slide to change, then navigate to a different view and go back to the home view. The carousel slide flashes back to the first slide.

An interesting side effect is that the highlighted navigation "dot" on the bottom of the carousel gets misplaced and no longer corresponds to the same slide. See screenshots below how active dot changes position before/after navigating to a different view.

img_3979
img_3980

### PR DESCRIPTION
**Pull request by jorystiefel**
Fixes #746.

Notable changes:
* Fixes issue of hero carousel not automatically advancing to next slide on launch
* Hero units now update when the entire feed updates, instead of every time `viewWillAppear` is called (as discussed with @alloy)
* Several tests that were related to the `viewWillAppear` coupling have been removed as they no longer make sense
* Added new test to make sure the hero units update when the feed refreshes

### CODE REVIEW

**Comment by orta** [comment]
any chance you can take a stab at adding a test or two to ARTabContentView? Ideally covering my mistake of not doing the view will appear etc? 

Perhaps use two partial mock VCs and ensure that when switching between two VCs without animations all these methods get called:

* viewDidLoad
* ViewWillAppear
* viewDidAppear
* viewWillDissapear
* viewDidDissapear

**Comment by jorystiefel** [comment]
@Orta, absolutely will update

**Comment by jorystiefel** [comment]
@alloy should be all set now

File: Artsy_Tests/View_Controller_Tests/Hero_Unit/ARHeroUnitTests.m [hunk 1]
```diff
@@ -92,6 +92,14 @@ - (void)getHeroUnitsWithSuccess:(void (^)())success failure:(void (^)(NSError *e
     });
 
     describe(@"handleHeroUnits", ^{
+        it(@"is called when feed updates", ^{
+            sharedBefore();
+            id mock = [OCMockObject partialMockForObject:heroVC];
+            [[mock expect] handleHeroUnits:heroUnits];
+            [heroVC.heroUnitNetworkModel setHeroUnits:heroUnits];
+            [mock verify];
+        });
```
Comment (alloy) [inline Artsy_Tests/View_Controller_Tests/Hero_Unit/ARHeroUnitTests.m]: This is not actually showing that the hero units update when the feed does. I’d like to see this move to the feed tests and show it happening as a part of the actual feed updating.

**Comment by alloy** [comment]
Tested this locally. All tests pass and seems to work good on both sim and device :+1: 

I’m gonna merge this, so can you please do that test fixing that I asked for in a separate PR?

File: Artsy_Tests/View_Controller_Tests/Hero_Unit/ARHeroUnitTests.m (hunk 1 above)
Comment (jorystiefel) [inline Artsy_Tests/View_Controller_Tests/Hero_Unit/ARHeroUnitTests.m]: No problem

**Pull request merged**